lib/mozart_fetcher/http_client.ex

```diff
@@ -39,6 +39,40 @@ defmodule HTTPClient do
     client.get(endpoint, headers, options)
   end
 
+  defp handle_response({:ok, response}) do
+    {:ok,
+     %HTTPoison.Response{
+       headers: response.headers,
+       body:
+         decode_response_body(response.body, content_encoding(process_headers(response.headers))),
+       status_code: response.status_code
+     }}
+  end
+
+  defp handle_response(response), do: response
+
+  defp process_headers(headers) do
+    Enum.map(headers, fn {k, v} -> {String.downcase(k), v} end)
```

Comment:
If we are using http1, does the casing of the response headers matter?

Reply:
These are response headers, so I think it's just normalising them for this line: https://github.com/bbc/mozart-fetcher/pull/46/files#diff-d515d4ebcd8ec474ea1d0cfcdc0c96c7R73

Reply:
It matters when I am looking for a specific header.

I see HTTPoison has callbacks for response headers and body, but there didn't seem to be one for both, so I left that out for now.